demos/device_defender_for_aws/defender_demo.c

```diff
@@ -76,6 +76,13 @@
  */
 #define DELAY_BETWEEEN_DEMO_ATTEMPTS_TICKS    ( pdMS_TO_TICKS( 5000U ) )
 
+/**
+ * @brief Size of the task numbers array.
+ *
+ * This must be at least the number of tasks used.
```

Comment:
Can you provide more explanation about why it should be the number of tasks used? What problem occurs when the size used for the list is less than the total number of tasks in the FreeRTOS system?

Reply:
I see that the [`vTaskGetInfo`](https://freertos.org/vTaskGetInfo.html) API does not take a length parameter for the number of tasks. Instead of pre-allocating the array, can we dynamically allocate the array based on the number of tasks running in the system?

Reply:
Not sure if it would be valuable. The array is not large, and just needs to be reasonably large. Updated the error to let the user know which number to increase.